client/signup/utils.js

```diff
@@ -177,7 +177,13 @@ export function getDesignTypeForSiteGoals( siteGoals, flow ) {
 
 export function getFilteredSteps( flowName, progress ) {
 	const flow = flows.getFlow( flowName );
-	return filter( progress, step => includes( flow.steps, step.stepName ) );
+
+	return sortBy(
+		// filter steps...
+		filter( progress, step => includes( flow.steps, step.stepName ) ),
+		// then order according to the flow definition...
+		( { stepName } ) => flow.steps.indexOf( stepName )
```

Comment:
What would happen if we started our journey in flow A but now we're in flow B?

Reply:
Ahh well, if they have completed 3 steps from flow A, then switched to flow B and flow B shared 2 out the 3 steps from flow A, those 2 steps would be held in state (as is the 1 unshared step, actually) so when we call `getFilteredSteps` after switching to flow B 2 of it's steps will returned .